### ISSUE TITLE
Netatmo Binding: Nullpointer Exception (again)

### ISSUE THREAD

**Issue by nesys** [openhab/openhab#3242]
Hi folks,

as issue #1820, I've updated from 1.6.2 to 1.7.1 (Binding 1.7.1, also 1.8.0 snapshot) and now Netatmo Binding seems not working any more.

I've tried multiple times the procedure in the wiki to install the certificate, even if I didn't change the Java version.

Here some information, please let me know if I need to collect more data-

Thanks

Andrea

ariela@HAB:~$ java -version
java version "1.8.0_31"
Java(TM) SE Runtime Environment (build 1.8.0_31-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.31-b07, mixed mode)

ariela@HAB:~$ uname -a
Linux HAB 3.2.0-4-amd64 #1 SMP Debian 3.2.65-1+deb7u1 x86_64 GNU/Linux
ariela@HAB:~$ 

concurrent log messages:

2015-10-04 08:45:59.547 [ERROR] [.service.AbstractActiveService] - Error while executing background thread Netatmo Refresh Service
java.lang.NullPointerException: null
	at org.openhab.binding.netatmo.internal.NetatmoBinding.execute(NetatmoBinding.java:206) ~[na:na]
	at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:156) ~[na:na]
	at org.openhab.core.service.AbstractActiveService$RefreshThread.run(AbstractActiveService.java:173) ~[na:na]

**Comment by watou** [comment]
If I were to guess, it looks like [this line](https://github.com/openhab/openhab/blob/master/bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoBinding.java#L206) is assuming that a `deviceListResponse` has already been received and saved.  @clinique or @robnielsen, could you take a look?

**Comment by nesys** [comment]
for reference, I've also reset the keys, and manually (via https://dev.netatmo.com/doc/methods/devicelist?access_token=xxxxxx) it works like a charm.

No idea if this helps.

**Comment by watou** [comment]
I suggest a pull request would produce a WARN log message instead of the NullPointerException, but the maintainer(s) can address that.

**Comment by robnielsen** [comment]
I'll see if I can figure out what's going on with this. Works just fine with  https://github.com/openhab/openhab/pull/2891, but that is running off a codebase that is about 3 months old.

**Comment by robnielsen** [comment]
@nesys can you include more of your log file. I'd like to see what's logged before the Null pointer exception. Does it happen right away, or after running for awhile.

**Comment by nesys** [comment]
what do you need exactly? I'm completely at your disposal

Andrea

**Comment by robnielsen** [comment]
Can you turn on netatmo debugging and attach your openhab.log file? Add the following to logback.xml: 

<logger name="org.openhab.binding.netatmo" level="DEBUG" />
<logger name="org.openhab.binding.netatmo.internal.messages" level="DEBUG" />

you might want to edit the file and x out the clientId,clientSecret, and refreshToken.

2015-10-04 12:11:54.927 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: RefreshTokenRequest[clientId=x,clientSecret=x,refreshToken=x]

**Comment by nesys** [comment]
what I need to add exactly in logback?

**Comment by robnielsen** [comment]
<logger name="org.openhab.binding.netatmo" level="DEBUG" />
    <logger name="org.openhab.binding.netatmo.internal.messages" level="DEBUG" />

**Comment by robnielsen** [comment]
Could you also try the Netatmo binding at https://openhab-ci.innoq.io//jenkins/job/openhab-pr-builder/43/ It also has my latest update

**Comment by nesys** [comment]
maybe that "moduleId=null" sounds not really good ... mmm ....

here the log:

2015-10-04 22:16:55.429 [INFO ] [.o.core.internal.CoreActivator] - openHAB runtime has been started (v1.7.1).
2015-10-04 22:16:57.607 [INFO ] [o.o.i.s.i.DiscoveryServiceImpl] - mDNS service has been started
2015-10-04 22:16:57.719 [INFO ] [o.o.i.s.i.DiscoveryServiceImpl] - Service Discovery initialization completed.
2015-10-04 22:16:57.754 [INFO ] [.io.transport.mqtt.MqttService] - MQTT Service initialization completed.
2015-10-04 22:16:57.760 [INFO ] [o.i.t.m.i.MqttBrokerConnection] - Starting MQTT broker connection 'mosquitto'
2015-10-04 22:16:59.773 [WARN ] [cpr.DefaultAnnotationProcessor] - Unable to detect annotations. Application may fail to deploy.
2015-10-04 22:17:01.391 [INFO ] [penhab.io.rest.RESTApplication] - Started REST API at /rest
2015-10-04 22:17:06.822 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'demo.sitemap'
2015-10-04 22:17:07.425 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'rrd4j.persist'
2015-10-04 22:17:07.464 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'exec.persist'
2015-10-04 22:17:07.475 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'logging.persist'
2015-10-04 22:17:07.494 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'myopenhab.persist'
2015-10-04 22:17:07.497 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'db4o.persist'
2015-10-04 22:17:07.579 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'demo.script'
2015-10-04 22:17:07.758 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'demo.items'
2015-10-04 22:17:08.189 [INFO ] [.o.u.w.i.servlet.WebAppServlet] - Started Classic UI at /openhab.app
2015-10-04 22:17:11.446 [INFO ] [.service.AbstractActiveService] - Sonos Refresh Service has been started
2015-10-04 22:17:11.462 [DEBUG] [.b.n.internal.NetatmoActivator] - Netatmo binding has been started.
2015-10-04 22:17:11.464 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#02:00:00:01:f2:5c#Temperature'
2015-10-04 22:17:11.465 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=02:00:00:01:f2:5c, measure=Temperature]
2015-10-04 22:17:11.471 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#02:00:00:01:f2:5c#Humidity'
2015-10-04 22:17:11.474 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=02:00:00:01:f2:5c, measure=Humidity]
2015-10-04 22:17:11.477 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#02:00:00:01:f2:5c#Rfstatus'
2015-10-04 22:17:11.478 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=02:00:00:01:f2:5c, measure=RfStatus]
2015-10-04 22:17:11.478 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#02:00:00:01:f2:5c#Batteryvp'
2015-10-04 22:17:11.478 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=02:00:00:01:f2:5c, measure=BatteryVp]
2015-10-04 22:17:11.478 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#Temperature'
2015-10-04 22:17:11.478 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=null, measure=Temperature]
2015-10-04 22:17:11.478 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#Humidity'
2015-10-04 22:17:11.479 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=null, measure=Humidity]
2015-10-04 22:17:11.479 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#CO2'
2015-10-04 22:17:11.480 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=null, measure=CO2]
2015-10-04 22:17:11.480 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#Pressure'
2015-10-04 22:17:11.480 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=null, measure=Pressure]
2015-10-04 22:17:11.481 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#Noise'
2015-10-04 22:17:11.484 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=null, measure=Noise]
2015-10-04 22:17:11.485 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#Wifistatus'
2015-10-04 22:17:11.486 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=null, measure=WifiStatus]
2015-10-04 22:17:11.486 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:1e:c8#Temperature'
2015-10-04 22:17:11.486 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:1e:c8, measure=Temperature]
2015-10-04 22:17:11.491 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:1e:c8#Humidity'
2015-10-04 22:17:11.491 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:1e:c8, measure=Humidity]
2015-10-04 22:17:11.491 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:1e:c8#CO2'
2015-10-04 22:17:11.492 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:1e:c8, measure=CO2]
2015-10-04 22:17:11.493 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:1e:c8#Rfstatus'
2015-10-04 22:17:11.493 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:1e:c8, measure=RfStatus]
2015-10-04 22:17:11.493 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:1e:c8#Batteryvp'
2015-10-04 22:17:11.493 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:1e:c8, measure=BatteryVp]
2015-10-04 22:17:11.496 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:24:2e#Temperature'
2015-10-04 22:17:11.496 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:24:2e, measure=Temperature]
2015-10-04 22:17:11.496 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:24:2e#Humidity'
2015-10-04 22:17:11.496 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:24:2e, measure=Humidity]
2015-10-04 22:17:11.497 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:24:2e#CO2'
2015-10-04 22:17:11.497 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:24:2e, measure=CO2]
2015-10-04 22:17:11.497 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:24:2e#Rfstatus'
2015-10-04 22:17:11.497 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:24:2e, measure=RfStatus]
2015-10-04 22:17:11.497 [DEBUG] [.NetatmoGenericBindingProvider] - Processing binding configuration: '70:ee:50:01:c2:66#03:00:00:01:24:2e#Batteryvp'
2015-10-04 22:17:11.501 [DEBUG] [.NetatmoGenericBindingProvider] - Adding binding: NetatmoBindingConfig [userid=null, deviceId=70:ee:50:01:c2:66, moduleId=03:00:00:01:24:2e, measure=BatteryVp]
2015-10-04 22:17:11.534 [INFO ] [.service.AbstractActiveService] - Netatmo Refresh Service has been started
2015-10-04 22:17:11.534 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Querying Netatmo API
2015-10-04 22:17:11.535 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Refreshing access token.
2015-10-04 22:17:11.789 [WARN ] [cpr.DefaultAnnotationProcessor] - Unable to detect annotations. Application may fail to deploy.
2015-10-04 22:17:11.803 [INFO ] [.b.sonos.internal.SonosBinding] - Querying the network for a predefined Sonos device with UDN uuid:RINCON_000E58289D5001400
2015-10-04 22:17:11.845 [INFO ] [.b.sonos.internal.SonosBinding] - Querying the network for any other Sonos device
2015-10-04 22:17:12.088 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: RefreshTokenRequest[clientId=xxxx,clientSecret=xxxx,refreshToken=xxxx]
2015-10-04 22:17:13.393 [INFO ] [.o.io.habmin.HABminApplication] - Started HABmin REST API at /services/habmin
2015-10-04 22:17:13.426 [INFO ] [.b.sonos.internal.SonosBinding] - Found a Sonos device (ZP90) with UDN uuid:RINCON_000E58289D5001400
2015-10-04 22:17:13.665 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: RefreshTokenResponse[accessToken=xxxx]
2015-10-04 22:17:14.808 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: DeviceListRequest[accessToken=xxxx]
2015-10-04 22:17:14.814 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: DeviceListResponse[status=ok,body=DeviceListResponse.Body[devices=[DeviceListResponse.Device[id=70:ee:50:01:c2:66,firmware=111,ip=<null>,lastFirmwareUpdate=<null>,lastRadioStore=<null>,lastStatusStore=Sat Jan 17 18:06:29 CET 1970,lastUpgrade=Sat Jan 17 17:01:48 CET 1970,moduleName=Living,modules=[02:00:00:01:f2:5c, 03:00:00:01:1e:c8, 03:00:00:01:24:2e],place=DeviceListResponse.Place[altitude=<null>,country=IT,location=[9.1878004074097, 45.461200714111],timezone=Europe/Rome,trustedLocation=<null>],publicData=<null>,stationName=HA-Netatmo.1,type=NAMain,owner=<null>,wifistatus=46]],modules=[DeviceListResponse.Module[id=02:00:00:01:f2:5c,firmware=43,mainDevice=70:ee:50:01:c2:66,moduleName=Outdoor,publicData=<null>,rfStatus=63,type=NAModule1], DeviceListResponse.Module[id=03:00:00:01:1e:c8,firmware=43,mainDevice=70:ee:50:01:c2:66,moduleName=Kitchen,publicData=<null>,rfStatus=79,type=NAModule4], DeviceListResponse.Module[id=03:00:00:01:24:2e,firmware=43,mainDevice=70:ee:50:01:c2:66,moduleName=Attic,publicData=<null>,rfStatus=78,type=NAModule4]]],executionTime=0.29449105262756]
2015-10-04 22:17:17.345 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,moduleId=02:00:00:01:f2:5c,measures=[Humidity, Temperature]]
2015-10-04 22:17:17.349 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[83, 17.1]],begTime=1443989439]]]
2015-10-04 22:17:17.827 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,measures=[CO2, Humidity, Noise, Pressure, Temperature]]
2015-10-04 22:17:17.830 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[1181, 74, 49, 1006.3, 21]],begTime=1443989473]]]
2015-10-04 22:17:17.986 [INFO ] [c.internal.ModelRepositoryImpl] - Loading model 'demo.rules'
2015-10-04 22:17:18.543 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,moduleId=03:00:00:01:1e:c8,measures=[CO2, Humidity, Temperature]]
2015-10-04 22:17:18.543 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[1885, 68, 20.9]],begTime=1443989439]]]
2015-10-04 22:17:19.197 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,moduleId=03:00:00:01:24:2e,measures=[CO2, Humidity, Temperature]]
2015-10-04 22:17:19.197 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[1269, 71, 20.1]],begTime=1443989445]]]
2015-10-04 22:17:19.219 [ERROR] [.service.AbstractActiveService] - Error while executing background thread Netatmo Refresh Service
java.lang.NullPointerException: null
	at org.openhab.binding.netatmo.internal.NetatmoBinding.execute(NetatmoBinding.java:206) ~[na:na]
	at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:156) ~[na:na]
	at org.openhab.core.service.AbstractActiveService$RefreshThread.run(AbstractActiveService.java:173) ~[na:na]
2015-10-04 22:21:19.037 [WARN ] [internal.ManagerConnectionImpl] - Unsupported protocol version 'Asterisk Call Manager/1.3'. Use at your own risk.
2015-10-04 22:21:19.988 [INFO ] [.service.AbstractActiveService] - HTTP Refresh Service has been started
2015-10-04 22:21:20.275 [INFO ] [.service.AbstractActiveService] - Exec Refresh Service has been started
2015-10-04 22:21:21.832 [WARN ] [.denon.internal.DenonConnector] - Encoding error in get
javax.xml.bind.UnmarshalException: null
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.handleStreamException(UnmarshallerImpl.java:468) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:402) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:371) ~[na:1.8.0_31]
	at org.openhab.binding.denon.internal.DenonConnector.getDocument(DenonConnector.java:549) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.setConfigProperties(DenonConnector.java:474) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.getInitialState(DenonConnector.java:189) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.connect(DenonConnector.java:147) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonBinding.updated(DenonBinding.java:189) [bundlefile:na]
	at org.eclipse.equinox.internal.cm.ManagedServiceTracker$1.run(ManagedServiceTracker.java:183) [org.eclipse.equinox.cm_1.0.400.v20120522-1841.jar:na]
	at org.eclipse.equinox.internal.cm.SerializedTaskQueue$1.run(SerializedTaskQueue.java:36) [org.eclipse.equinox.cm_1.0.400.v20120522-1841.jar:na]
Caused by: com.sun.istack.internal.SAXParseException2: unexpected element (uri:"", local:"html"). Expected elements are <{}device_Info>
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:726) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:247) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:242) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:109) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1131) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:556) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:538) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.InterningXmlVisitor.startElement(InterningXmlVisitor.java:60) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:231) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:165) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:400) ~[na:1.8.0_31]
	... 8 common frames omitted
Caused by: javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}device_Info>
	... 19 common frames omitted
2015-10-04 22:21:21.939 [INFO ] [org.openhab.model.script.Rules] - UV Index Rating completed
2015-10-04 22:21:22.086 [WARN ] [.denon.internal.DenonConnector] - Encoding error in post
javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}cmd>,<{}rx>,<{}text>
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:726) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:247) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:242) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:109) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1131) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:556) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:538) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.SAXConnector.startElement(SAXConnector.java:153) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:509) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.scanStartElement(XMLNSDocumentScannerImpl.java:379) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl$NSContentDriver.scanRootElementHook(XMLNSDocumentScannerImpl.java:605) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:3138) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:880) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:117) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1213) ~[na:1.8.0_31]
	at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:649) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:243) ~[na:1.8.0_31]
	at com.sun.xml.internal.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:214) ~[na:1.8.0_31]
	at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:157) ~[na:1.8.0_31]
	at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:204) ~[na:1.8.0_31]
	at org.openhab.binding.denon.internal.DenonConnector.postDocument(DenonConnector.java:573) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.updateDisplayInfo(DenonConnector.java:460) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.refreshHttpProperties(DenonConnector.java:496) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.updateState(DenonConnector.java:199) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.getInitialState(DenonConnector.java:190) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonConnector.connect(DenonConnector.java:147) [bundlefile:na]
	at org.openhab.binding.denon.internal.DenonBinding.updated(DenonBinding.java:189) [bundlefile:na]
	at org.eclipse.equinox.internal.cm.ManagedServiceTracker$1.run(ManagedServiceTracker.java:183) [org.eclipse.equinox.cm_1.0.400.v20120522-1841.jar:na]
	at org.eclipse.equinox.internal.cm.SerializedTaskQueue$1.run(SerializedTaskQueue.java:36) [org.eclipse.equinox.cm_1.0.400.v20120522-1841.jar:na]
2015-10-04 22:21:22.117 [INFO ] [.service.AbstractActiveService] - NTP Refresh Service has been started
2015-10-04 22:21:22.124 [INFO ] [.service.AbstractActiveService] - ZWave Refresh Service has been started
2015-10-04 22:21:22.652 [INFO ] [.service.AbstractActiveService] - Nest Refresh Service has been started
2015-10-04 22:21:22.658 [INFO ] [.b.a.internal.bus.AstroBinding] - AstroConfig[latitude=45.464902,longitude=9.094843,interval=disabled,systemTimezone=Europe/Rome (CET +0200),daylightSavings=true]
2015-10-04 22:21:23.337 [INFO ] [.myopenhab.internal.MyOHClient] - Connected to my.openHAB service
2015-10-04 22:21:25.157 [INFO ] [org.openhab.model.script.Rules] - PreciProb rounding completed
2015-10-04 22:21:25.582 [INFO ] [org.openhab.model.script.Rules] - Ozone String to Number completed
2015-10-04 22:21:25.683 [INFO ] [.b.a.internal.job.JobScheduler] - Scheduled a daily job at midnight for astro calculation
2015-10-04 22:21:25.690 [WARN ] [.b.a.internal.job.JobScheduler] - Sun azimuth/elevation and/or moon distance/illumination binding available, but configuration is disabled (interval = 0)!
2015-10-04 22:21:25.947 [INFO ] [org.openhab.model.script.Rules] - Yahoo Wind Compass completed
2015-10-04 22:21:26.432 [INFO ] [org.openhab.model.script.Rules] - CloudCover rounding completed
2015-10-04 22:05:51.087 [INFO ] [.b.a.internal.job.JobScheduler] - Scheduled a daily job at midnight for astro calculation
2015-10-04 22:05:51.095 [WARN ] [.b.a.internal.job.JobScheduler] - Sun azimuth/elevation and/or moon distance/illumination binding available, but configuration is disabled (interval = 0)!
2015-10-04 22:07:41.047 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Querying Netatmo API
2015-10-04 22:07:41.927 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,moduleId=02:00:00:01:f2:5c,measures=[Humidity, Temperature]]
2015-10-04 22:07:41.928 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[83, 17.2]],begTime=1443988825]]]
2015-10-04 22:07:44.117 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,measures=[CO2, Humidity, Noise, Pressure, Temperature]]
2015-10-04 22:07:44.118 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[1187, 74, 37, 1006.3, 21]],begTime=1443988872]]]
2015-10-04 22:07:44.842 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,moduleId=03:00:00:01:1e:c8,measures=[CO2, Humidity, Temperature]]
2015-10-04 22:07:44.845 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[1752, 68, 20.9]],begTime=1443988825]]]
2015-10-04 22:07:45.864 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Request: MeasurementRequest[accessToken=xxxx,deviceId=70:ee:50:01:c2:66,moduleId=03:00:00:01:24:2e,measures=[CO2, Humidity, Temperature]]
2015-10-04 22:07:45.865 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: MeasurementResponse[status=ok,body=[MeasurementResponse.Body[values=[[1287, 71, 20.1]],begTime=1443988831]]]
2015-10-04 22:07:45.874 [ERROR] [.service.AbstractActiveService] - Error while executing background thread Netatmo Refresh Service
java.lang.NullPointerException: null
	at org.openhab.binding.netatmo.internal.NetatmoBinding.execute(NetatmoBinding.java:206) ~[na:na]
	at org.openhab.core.binding.AbstractActiveBinding$BindingActiveService.execute(AbstractActiveBinding.java:156) ~[na:na]
	at org.openhab.core.service.AbstractActiveService$RefreshThread.run(AbstractActiveService.java:173) ~[na:na]

**Comment by nesys** [comment]
sorry if there is another issue with another Binding (Denon) .... but I've pasted the log as it is

**Comment by nesys** [comment]
what you suggest? the "Tests" version, or the other?

**Comment by robnielsen** [comment]
https://openhab-ci.innoq.io//jenkins/job/openhab-pr-builder/43/org.openhab.binding$org.openhab.binding.netatmo/artifact/org.openhab.binding/org.openhab.binding.netatmo/1.8.0-SNAPSHOT/org.openhab.binding.netatmo-1.8.0-SNAPSHOT.jar

**Comment by nesys** [comment]
installed, sounds exactly the same as 1.7.1

**Comment by robnielsen** [comment]
Would you be willing to share your configuration info with me and i can try to debug locally? My email address is rob[dot]nielsen[at]yahoo.com?

**Comment by nesys** [comment]
Yes sure. What files do you need?

**Comment by robnielsen** [comment]
I think I see what is going on...

    2015-10-04 22:17:14.814 [DEBUG] [.o.b.n.internal.NetatmoBinding] - Response: DeviceListResponse[status=ok,body=DeviceListResponse.Body[devices=[DeviceListResponse.Device[id=70:ee:50:01:c2:66,firmware=111,ip=,lastFirmwareUpdate=,lastRadioStore=,lastStatusStore=Sat Jan 17 18:06:29 CET 1970,lastUpgrade=Sat Jan 17 17:01:48 CET 1970,moduleName=Living,modules=[02:00:00:01:f2:5c, 03:00:00:01:1e:c8, 03:00:00:01:24:2e],place=DeviceListResponse.Place[altitude=,country=IT,location=[9.1878004074097, 45.461200714111],timezone=Europe/Rome,trustedLocation=],publicData=,stationName=HA-Netatmo.1,type=NAMain,owner=,wifistatus=46]],modules=[DeviceListResponse.Module[id=02:00:00:01:f2:5c,firmware=43,mainDevice=70:ee:50:01:c2:66,moduleName=Outdoor,publicData=,rfStatus=63,type=NAModule1], DeviceListResponse.Module[id=03:00:00:01:1e:c8,firmware=43,mainDevice=70:ee:50:01:c2:66,moduleName=Kitchen,publicData=,rfStatus=79,type=NAModule4], DeviceListResponse.Module[id=03:00:00:01:24:2e,firmware=43,mainDevice=70:ee:50:01:c2:66,moduleName=Attic,publicData=,rfStatus=78,type=NAModule4]]],executionTime=0.29449105262756]

all the values returned are key=value, and the place it's getting a null pointer exception is from the altitude. If you look above, you'll see altitude=,

**Comment by nesys** [comment]
is it an issue with my config? what can I do?

**Comment by robnielsen** [comment]
Can you send me your
netatmo:clientid
netatmo:clientsecret
netatmo:refreshtoken

And then I can verify. What's the altitude that shows up at https://my.netatmo.com

**Comment by nesys** [comment]
<img width="1269" alt="screen shot 2015-10-04 at 23 49 50" src="https://cloud.githubusercontent.com/assets/8050464/10270468/caba883a-6af2-11e5-8f66-b812fa4eaf6f.png">

altitude is null :(  sending info

**Comment by robnielsen** [comment]
That was it, it is fixed in https://github.com/openhab/openhab/pull/2891

**Comment by nesys** [comment]
thanks rob. It should be included in a snapshot?

**Comment by robnielsen** [comment]
@teichsta, can https://github.com/openhab/openhab/pull/2891 get merged? Thanks!

**Comment by watou** [comment]
@robnielson, please create a separate PR for this fix. You must be unaware that combining unrelated changes in a single PR is problematic. :-). @teichsta, please don't merge #2891 until those old issues are resolved.

**Comment by clinique** [comment]
Nice finding - I will report this in the v2.0 of the binding

**Comment by robnielsen** [comment]
@watou, there are no issues with the PR request, other than you don't like the level of precision that was picked. It meets the standard practices of precision used by both the scientific and engineering community. No different than you did with the ecobee binding, you picked a level of precision also.

**Comment by watou** [comment]
I disagree with your characterization, @robnielsen.  You were instructed that "[o]nly the UI should decide how many digits to show" and despite clear explanations as to why this is preferred, you chose to round off precision in item states to the detriment of accuracy, leading to bad data when computing totals, averages, etc.  Separately, combining discrete fixes into an unrelated enhancement complicates code management.  I ask again that discrete bug fixes be placed in their own PRs.  Thanks, Rob.

**Comment by nesys** [comment]
folks, I didn't follow the thread as appropriate, but ... it should be fixed in 1.8.0?

Thank you very much for your support

**Comment by robnielsen** [comment]
The fix is now merged into 1.8.0

**Comment by teichsta** [comment]
fixed with #2891
